app/javascript/dashboard/components/ConversationItem.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+  <conversation-card
+    :key="source.id"
+    :active-label="label"
+    :team-id="teamId"
+    :folders-id="foldersId"
+    :chat="source"
+    :conversation-type="conversationType"
+    :selected="isConversationSelected(source.id)"
+    :show-assignee="showAssignee"
+    :enable-context-menu="true"
+    @select-conversation="selectConversation"
+    @de-select-conversation="deSelectConversation"
+    @assign-agent="onAssignAgent"
+    @assign-team="onAssignTeam"
+    @assign-label="onAssignLabel"
+    @update-conversation-status="toggleConversationStatus"
+    @context-menu-toggle="onContextMenuToggle"
+    @mark-as-unread="markAsUnread"
+    @assign-priority="assignPriority"
+  />
+</template>
+
+<script>
+import ConversationCard from './widgets/conversation/ConversationCard.vue';
+export default {
+  components: {
+    ConversationCard,
+  },
+  props: {
+    source: {
+      type: Object,
+      required: true,
+    },
+    teamId: {
+      type: [String, Number],
+      default: 0,
+    },
+    label: {
+      type: String,
+      default: '',
+    },
+    conversationType: {
+      type: String,
+      default: '',
+    },
+    foldersId: {
+      type: [String, Number],
+      default: 0,
+    },
+    isConversationSelected: {
+      type: Function,
+      default: () => {},
+    },
+    showAssignee: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  methods: {
+    selectConversation(conversationId, inboxId) {
+      this.$parent.$parent.$emit(
```

Comment:
Using `$parent.$parent` is a bad idea, it also makes this component tightly coupled to the component tree. It's better to use the global event bus. Even better would be to use `provide` and `inject`

In the parent

```js
export default {
  provide() {
    return {
      assignAgent: this.assignAgent,
      assignTeam: this.assignTeam,
      assignLabel: this.assignLabel,
      updateConversationStatus: this.updateConversationStatus,
      toggleContextMenu: this.toggleContextMenu,
      markAsUnread: this.markAsUnread,
      assignPriority: this.assignPriority
    };
  },
}
```

and in this component, you can have 

```js
// In child component
export default {
  inject: ['assignAgent', 'assignTeam', 'assignLabel', 'updateConversationStatus', 'toggleContextMenu', 'markAsUnread', 'assignPriority'],
  methods: {
    onAssignAgent(agent, conversationId) {
      this.assignAgent(agent, conversationId);
    },
    // Define other methods similarly...
  }
};
```

Reply:
Sounds good will update it.

Reply:
@scmmishra Done https://github.com/chatwoot/chatwoot/pull/8540/commits/8ad9d44910d6c55e27131ea335d82f5f9a0c03a2

Reply:
Used `provide ` and `inject`